lib/android-helpers.js

```diff
@@ -372,39 +373,65 @@ helpers.pushStrings = async function (language, adb, opts) {
   return {};
 };
 
-helpers.unlock = async function (adb) {
+helpers.unlockWithUIAutomation = async function (driver, adb, unlockCapabilities) {
+  let unlockType = unlockCapabilities.unlockType;
+  if (!unlocker.isValidUnlockType(unlockType)) {
+    throw new Error(`Invalid unlock type ${unlockType}`);
+  }
+  let unlockKey = unlockCapabilities.unlockKey;
+  if (!unlocker.isValidKey(unlockType, unlockKey)) {
+    throw new Error(`Missing unlockKey ${unlockKey} capability for unlockType ${unlockType}`);
+  }
+  const unlockMethod = {
+    [PIN_UNLOCK]: unlocker.pinUnlock,
+    [PASSWORD_UNLOCK]: unlocker.passwordUnlock,
+    [PATTERN_UNLOCK]: unlocker.patternUnlock,
+  }[unlockType];
+  await unlockMethod(adb, driver, unlockCapabilities);
+};
+
+helpers.unlockWithHelperApp = async function(adb) {
+  logger.info("Unlocking screen");
+  await adb.forceStop('io.appium.unlock');
+  // then start the app twice, as once is flakey
+  let startOpts = {
+    pkg: "io.appium.unlock",
+    activity: ".Unlock",
+    action: "android.intent.action.MAIN",
+    category: "android.intent.category.LAUNCHER",
+    flags: "0x10200000",
+    stopApp: false
+  };
+  await adb.startApp(startOpts);
+  await adb.startApp(startOpts);
+};
+
+helpers.unlock = async function (driver, adb, capabilities) {
   if (!(await adb.isScreenLocked())) {
     logger.info("Screen already unlocked, doing nothing");
     return;
   }
-  logger.info("Unlocking screen");
-
-  await retryInterval(10, 1000, async () => {
-    logger.debug("Screen is locked, trying to unlock");
-
-    // first manually stop the unlock activity
-    await adb.forceStop('io.appium.unlock');
-
-    // then start the app twice, as once is flakey
-    let startOpts = {
-      pkg: "io.appium.unlock",
-      activity: ".Unlock",
-      action: "android.intent.action.MAIN",
-      category: "android.intent.category.LAUNCHER",
-      flags: "0x10200000",
-      stopApp: false
-    };
-    await adb.startApp(startOpts);
-    await adb.startApp(startOpts);
-
-    // check if it worked, twice
-    await retryInterval(2, 1000, async () => {
-      if (!await adb.isScreenLocked()) {
-        logger.debug("Screen unlocked successfully");
-      } else {
-        throw new Error("Screen did not unlock successfully, retrying");
-      }
+  if (typeof capabilities.unlockType === 'undefined') {
```

Comment:
we used `_.isUndefined` below for this same purpose, maybe we should do that here to be consistent?